File: python/algorithms/sieveEratosthenes.py
# IMPLEMENTATION OF SIEVE OF ERATOSTHENES TO FIND PRIME NUMBERS BEFORE A NUMBER N
# IN PYTHON 3

def sieveEratosthenes(n):
    prime = [True for i in range(n+1)]              #make a bool array of size n 
    p = 2 
    while(p*p <= n):
        if prime[p]==True:                          
            for i in range(p*p,n+1,p):              #append by p to eliminate all multiples of p
                prime[i]=False                      
                
        p+=1
    
    for p in range(2,n):
        if prime[p]:                                #if prime[p] is not false
            print(p)
            

n = int(input())
print(sieveEratosthenes(n))

File: c/sieveEratosthenes.c
#include<stdio.h>

//Added a typedef enum with object bool to implement bool,
// since it isn't present in C.

typedef enum 
{
    false, true
}bool;



void SieveOfEratosthenes(int n) 
{ 
    // Create a boolean array "prime[0..n]" and initialize 
    // all entries it as true. A value in prime[i] will 
    // finally be false if i is Not a prime, else true. 
    bool prime[n+1]; 
    for(int i = 0;i<n+1;i++)
    {        
        prime[i]=true;
    }
    for (int p=2; p*p<=n; p++) 
    { 
        // If prime[p] is not changed, then it is a prime 
        if (prime[p] == true) 
        { 
            // Eliminate all multiples of p greater than or  
            // equal to the square of it since 
            // numbers which are multiple of p and are 
            // less than its square are assigned true initially and are primes
            // since they do not have common factors with previous numbers  
            for (int i=p*p; i<=n; i += p) 
                prime[i] = false; 
        } 
    } 
  
    // Print all prime numbers 
    for (int p=2; p<=n; p++) 
       if (prime[p]!=false) 
            printf("%d ", p); 
} 
  


// Driver Program to test above function 
int main() 
{ 
    int n;

    printf("\n\n\tEnter the number to find the prime numbers preceding it: ");
    scanf("%d",&n); 
    
    printf("\n\n\tThe following are the prime numbers smaller than or equal to %d:\n\n\t", n);
    SieveOfEratosthenes(n); 
    
    printf("\n");

    return 0; 
} 